opendevin/core/main.py

```diff
@@ -85,9 +85,12 @@ async def main(
     agent = AgentCls(llm=llm)
 
     event_stream = EventStream('main')
+
+    max_iterations = min(args.max_iterations, args.global_max_iterations)
+
     controller = AgentController(
         agent=agent,
-        max_iterations=args.max_iterations,
+        max_iterations=max_iterations,
```

Comment:
I am little confuse why create `global_max_iterations` when we already have `args.max_iterations`.

Reply:
https://github.com/OpenDevin/OpenDevin/issues/2121